inc/class-upgrade-history.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Internal
+ */
+
+/**
+ * This class handles storing the current options for future reference.
+ *
+ * This should only be used during an upgrade routine.
+ */
+class WPSEO_Upgrade_History {
+
+	/**
+	 * Option to use to store/retrieve data from.
+	 *
+	 * @var string
+	 */
+	protected $option_name = 'wpseo_upgrade_history';
+
+	/**
+	 * WPSEO_Upgrade_History constructor.
+	 *
+	 * @param null|string $option_name Optional. Custom option to use to store/retrieve history from.
+	 */
+	public function __construct( $option_name = null ) {
+		if ( $option_name !== null ) {
+			$this->option_name = $option_name;
+		}
+	}
+
+	/**
+	 * Retrieves the content of the history items currently stored.
+	 *
+	 * @return array The contents of the history option.
+	 */
+	public function get() {
+		$data = get_option( $this->get_option_name(), array() );
+		if ( ! is_array( $data ) ) {
+			return array();
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Adds a new history entry in the storage.
+	 *
+	 * @param string $old_version  The version we are upgrading from.
+	 * @param string $new_version  The version we are upgrading to.
+	 * @param array  $option_names The options that need to be stored.
+	 */
+	public function add( $old_version, $new_version, array $option_names ) {
+		$option_data = array();
+		if ( array() !== $option_names ) {
+			$option_data = $this->get_options_data( $option_names );
+		}
+
+		// Retrieve current history.
+		$data = $this->get();
+
+		// Add new entry.
+		$data[ time() ] = array(
+			'options' => $option_data,
```

Comment:
WP core coding standards say this should be indented at the same level as the statements below, not sure whether this applies to Yoast coding standards too.